src/code/z_parameter.c

```diff
@@ -2779,17 +2800,552 @@ s16 D_801BFB24[] = { 0, 255, 100, 0 };   // magicArrowEffectsB
+
+                            colorStep =
+                                ABS_ALT(sFinalHoursClockFrameEnvBlue - D_801BFC10[sFinalHoursClockColorTargetIndex]) /
+                                sFinalHoursClockColorTimer;
+                            if (sFinalHoursClockFrameEnvBlue >= D_801BFC10[sFinalHoursClockColorTargetIndex]) {
+                                sFinalHoursClockFrameEnvBlue -= colorStep;
+                            } else {
+                                sFinalHoursClockFrameEnvBlue += colorStep;
+                            }
+
+                            sFinalHoursClockColorTimer--;
+
+                            if (sFinalHoursClockColorTimer == 0) {
+                                sFinalHoursClockDigitsRed = D_801BFC04[sFinalHoursClockColorTargetIndex];
+                                sFinalHoursClockFrameEnvRed = D_801BFC08[sFinalHoursClockColorTargetIndex];
+                                sFinalHoursClockFrameEnvGreen = D_801BFC0C[sFinalHoursClockColorTargetIndex];
+                                sFinalHoursClockFrameEnvBlue = D_801BFC10[sFinalHoursClockColorTargetIndex];
+                                sFinalHoursClockColorTimer = 6;
+                                sFinalHoursClockColorTargetIndex ^= 1;
+                            }
+                        }
+
+                        sp1E6 = sThreeDayClockAlpha;
+                        if (sp1E6 != 0) {
+                            sp1E6 = 255;
+                        }
+
+                        func_8012C654(play->state.gfxCtx);
+
+                        /**
+                         * Draws Final-Hours Clock's Frame
+                         */
+                        gSPMatrix(OVERLAY_DISP++, &gIdentityMtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                        gDPSetAlphaCompare(OVERLAY_DISP++, G_AC_THRESHOLD);
+                        gDPSetRenderMode(OVERLAY_DISP++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+                        gDPSetCombineLERP(OVERLAY_DISP++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0,
+                                          PRIMITIVE, 0, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0,
+                                          PRIMITIVE, 0);
+                        gDPSetPrimColor(OVERLAY_DISP++, 0, 0, 255, 255, 195, sp1E6);
+                        gDPSetEnvColor(OVERLAY_DISP++, sFinalHoursClockFrameEnvRed, sFinalHoursClockFrameEnvGreen,
+                                       sFinalHoursClockFrameEnvBlue, 0);
+
+                        OVERLAY_DISP = Gfx_DrawTexRect4b(OVERLAY_DISP, gFinalHoursClockFrameTex, 3, 80, 13, 119, 202,
+                                                         80, 13, 0, 0, 0, 1 << 10, 1 << 10);
+
+                        finalHoursClockSlots[0] = 0;
+
+                        timeBeforeMoonCrash = (4 - CURRENT_DAY) * DAY_LENGTH -
+                                              (u16)(((void)0, gSaveContext.save.time) - CLOCK_TIME(6, 0));
```

Comment:
This seems to be popping up quite frequently now, should we give it a macro now?

Reply:
Hmm yeah, we could consider it. Just something like `TIME_BEFORE_MOON_CRASH`? Or maybe `TIME_UNTIL_MOON_CRASH`?